diff --git a/hacks/Makefile.in b/hacks/Makefile.in
--- a/hacks/Makefile.in
+++ b/hacks/Makefile.in
@@ -268,7 +268,7 @@ TARFILES	= $(SRCS) $(HDRS) $(SCRIPTS) $(MEN) $(RETIRED_MEN) \
 
 
 default: all
-all: $(EXES) $(RETIRED_EXES)
+all: $(EXES)
 
 install:   install-program   install-scripts install-xml install-man
 uninstall: uninstall-program uninstall-xml uninstall-man
diff --git a/hacks/glx/Makefile.in b/hacks/glx/Makefile.in
--- a/hacks/glx/Makefile.in
+++ b/hacks/glx/Makefile.in
@@ -276,7 +276,7 @@ TARFILES	= $(SRCS) $(HDRS) $(GL_MEN) $(RETIRED_MEN) $(EXTRAS)
 
 
 default: all
-all: $(EXES) $(RETIRED_EXES)
+all: $(EXES)
 
 install:   install-program    install-xml   install-man
 uninstall: uninstall-program  uninstall-xml uninstall-man
