diff --git a/driver/Makefile.in b/driver/Makefile.in
--- a/driver/Makefile.in
+++ b/driver/Makefile.in
@@ -257,7 +257,7 @@ all: $(EXES) $(EXES2)
 tests: $(TEST_EXES)
 
 install:   install-program   install-ad    install-scripts \
-	   install-gnome     install-man   install-xml install-pam
+	   install-gnome     install-man   install-xml
 uninstall: uninstall-program uninstall-ad  \
 	   uninstall-gnome   uninstall-man uninstall-xml  
 
