diff --git a/driver/XScreenSaver.ad.in b/driver/XScreenSaver.ad.in
index fa2587a..ef3a17d 100644
--- a/driver/XScreenSaver.ad.in
+++ b/driver/XScreenSaver.ad.in
@@ -123,7 +123,9 @@ GetViewPortIsFullOfLies: False
 
 ! The format used for printing the date and time in the password dialog box
 ! (see the strftime(3) manual page for details.)
-*dateFormat:		%d-%b-%y (%a); %I:%M %p
+*dateFormat:		%Y-%m-%d %H:%M:%S
+! Original upstream format
+! *dateFormat:		%d-%b-%y (%a); %I:%M %p
 ! To show the time only:
 ! *dateFormat:		%I:%M %p
 ! For 24 hour time:
