diff --git a/driver/XScreenSaver.ad.in b/driver/XScreenSaver.ad.in
index fa2587a..a8f6bc8 100644
--- a/driver/XScreenSaver.ad.in
+++ b/driver/XScreenSaver.ad.in
@@ -253,7 +253,6 @@ GetViewPortIsFullOfLies: False
 				xspirograph -root			    \n\
 @GL_KLUDGE@ GL: 				circuit -root				    \n\
 @GL_KLUDGE@ GL: 				dangerball -root			    \n\
-- GL: 				dnalogo -root				    \n\
 @GL_KLUDGE@ GL: 				engine -root				    \n\
 @GL_KLUDGE@ GL: 				flipscreen3d -root			    \n\
 @GL_KLUDGE@ GL: 				gltext -root				    \n\
