diff --git a/driver/XScreenSaver.ad.in b/driver/XScreenSaver.ad.in
index fa2587a..ef753d8 100644
--- a/driver/XScreenSaver.ad.in
+++ b/driver/XScreenSaver.ad.in
@@ -186,6 +186,7 @@ GetViewPortIsFullOfLies: False
 				goop -root				    \n\
 				grav -root				    \n\
 				ifs -root				    \n\
+				unicode -root				    \n\
 @GL_KLUDGE@ GL: 				jigsaw -root				    \n\
 				julia -root				    \n\
 -				kaleidescope -root			    \n\
