diff --git a/driver/XScreenSaver.ad.in b/driver/XScreenSaver.ad.in
index fa2587a..4406551 100644
--- a/driver/XScreenSaver.ad.in
+++ b/driver/XScreenSaver.ad.in
@@ -454,10 +454,11 @@ XScreenSaver.bourneShell:		/bin/sh
 *Dialog.shadowThickness:	2
 
 *passwd.heading.label:		XScreenSaver %s
-*passwd.body.label:		This screen is locked.
+*passwd.body.label:		Please enter your password.
 *passwd.unlock.label:		OK
 *passwd.login.label:		New Login
 *passwd.user.label:		Username:
+*passwd.passwd.label:		Password:
 *passwd.thermometer.width:	8
 *passwd.asterisks:              True
 *passwd.uname:                  True
