diff --git a/Makefile.in b/Makefile.in
--- a/Makefile.in
+++ b/Makefile.in
@@ -17,6 +17,7 @@ TARFILES	= README README.hacking README.VMS INSTALL \
 		  OSX/xscreensaver.xcodeproj/project.pbxproj
 
 TAR		= tar
+SUBDIRS2	:= $(filter-out OSX android,$(SUBDIRS2))
 
 MAKE_SUBDIR  = for dir in $(SUBDIRS); do (cd $$dir; $(MAKE) $@) || exit 5; done
 MAKE_SUBDIR2 = for dir in $(SUBDIRS2); do (cd $$dir; $(MAKE) $@) || exit 5; done
