diff --git a/Makefile.in b/Makefile.in
index 17c698d..2e0117a 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -7,7 +7,7 @@ VPATH		= @srcdir@
 
 SHELL		= /bin/sh
 SUBDIRS 	= utils jwxyz hacks hacks/glx driver po
-SUBDIRS2	= $(SUBDIRS) OSX android
+SUBDIRS2	= $(SUBDIRS)
 TARFILES	= README README.hacking README.VMS INSTALL \
 		  configure configure.in Makefile.in config.h.in \
 		  config.h-vms install-sh setup.com config.guess aclocal.m4 \
