diff --git a/Makefile.in b/Makefile.in
index 17c698d..bfcf569 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -54,8 +54,8 @@ clean::
 	@$(MAKE_SUBDIR2)
 
 distclean:: clean
-	-rm -f config.h Makefile config.status config.cache config.log TAGS *~ "#"* intltool-extract intltool-merge intltool-update
 	@$(MAKE_SUBDIR2)
+	-rm -f config.h Makefile config.status config.cache config.log TAGS *~ "#"* intltool-extract intltool-merge intltool-update
 
 dist:: tar
 
